nodes/list_masks/mask_join.py

```diff
@@ -87,28 +103,29 @@ def apply_mask(self, mask, data_t, data_f):
         return out
 
     def get_level(self, mask, data_t, data_f, level):
+        if self.choice:
+            apply_mask = self.apply_choice_mask
+        else:
+            apply_mask = self.apply_mask
```

Comment:
It's not very important but I don't agree that this declaration should be moved to the start of the function. I like the idea that variables should declared as close as possible to place of their usage.
https://softwareengineering.stackexchange.com/questions/113262/why-declare-variables-close-to-where-they-are-used

Reply:
The `apply_mask` is used in both `if level == 1` and `else`, so, to declare it only once, it cannot be declared elsewhere.